storybookjs/storybook
# Core: Remove manager redirects on boot

Issue: Weird `setTimeout(..,1)` navigations in manager url module.

The reason was previously we used the manager URL to drive the story selection.

Now we simply use the storyId, which can be a prefix. We don't need to change the URL until the preview emits the `CURRENT_STORY_WAS_SET` event (at which point the manager sets the path).

## What I did

Dropped the redirects. Fixed up the support for legacy URLs.

## How to test

See tests.

Try the following URLs in official SB:

 - http://localhost:9011/ 
 - http://localhost:9011/?random-stuff=that-should-be-stripped
 - http://localhost:9011/?path=/story/addons-controls--basic
 - http://localhost:9011/?selectedKind=addons-actions&selectedStory=arg-types-example
 - http://localhost:9011/?selectedKind=addons-actions
